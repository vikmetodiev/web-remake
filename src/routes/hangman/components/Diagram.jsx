import React from 'react';
import styled from 'styled-components';
import png1 from '../img/1.png';
import png2 from '../img/2.png';
import png3 from '../img/3.png';
import png4 from '../img/4.png';
import png5 from '../img/5.png';
import png6 from '../img/6.png';
import png7 from '../img/7.png';

const Wrapper = styled.div`
  margin-top:-25px;
  width: 295px;
  height: 295px;
  background: url(${png1}) no-repeat center center;
  background-size: 100%;
  margin: 0 15px;
  display: ${props => (props.newGame ? 'block' : 'none')};
  opacity: ${props => (props.newGame ? 1 : 0)};
  transition: all 0.3s linear;
  &[data-order='2'] {
    background-image: url(${png2});
  }
  &[data-order='3'] {
    background-image: url(${png3});
  }
  &[data-order='4'] {
    background-image: url(${png4});
  }
  &[data-order='5'] {
    background-image: url(${png5});
  }
  &[data-order='6'] {
    background-image: url(${png6});
  }
  &[data-order='7'] {
    background-image: url(${png7});
  }
`;

const Diagram = props => (
  <Wrapper {...props} data-order={props.currentDiagram} className="diagram" />
);

export default Diagram;
