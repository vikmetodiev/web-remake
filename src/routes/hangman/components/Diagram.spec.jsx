import React from 'react';

import toJson from 'enzyme-to-json';
import Diagram from './Diagram';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('Diagram', () => {
  it('should render the component', () => {
    const wrapper = shallow(<Diagram />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
