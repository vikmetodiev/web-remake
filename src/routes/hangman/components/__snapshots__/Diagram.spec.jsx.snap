// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagram should render the component 1`] = `
<styled.div
  className="diagram"
/>
`;
